autofs-5.0.4 - libxml2 workaround fix

From: Ian Kent <raven@themaw.net>

Add a check for libxml2.so.2 for the libxml2 workaround in case libxml2.so
is not present.
---

 CHANGELOG          |    1 +
 daemon/automount.c |    2 ++
 2 files changed, 3 insertions(+), 0 deletions(-)


diff --git a/CHANGELOG b/CHANGELOG
index 1cb56fe..b8ad22e 100644
--- a/CHANGELOG
+++ b/CHANGELOG
@@ -21,6 +21,7 @@
 - update to provide ipv6 name and address support.
 - update to provide ipv6 address parsing.
 - easy alloca replacements fix.
+- add check for alternate libxml2 library for libxml2 tsd workaround.
 
 4/11/2008 autofs-5.0.4
 -----------------------
diff --git a/daemon/automount.c b/daemon/automount.c
index 1ec686b..a664277 100644
--- a/daemon/automount.c
+++ b/daemon/automount.c
@@ -2058,6 +2058,8 @@ int main(int argc, char *argv[])
 
 #ifdef LIBXML2_WORKAROUND
 	void *dh_xml2 = dlopen("libxml2.so", RTLD_NOW);
+	if (!dh_xml2)
+		dh_xml2 = dlopen("libxml2.so.2", RTLD_NOW);
 #endif
 #ifdef TIRPC_WORKAROUND
 	void *dh_tirpc = dlopen("libitirpc.so", RTLD_NOW);
